pkg/triggerconfig/config_types.go

```diff
@@ -54,18 +53,26 @@ type JobBase struct {
 	// If this field is unspecified or false, a new pod will be created to replace
 	// the evicted one.
 	ErrorOnEviction bool `json:"errorOnEviction,omitempty"`
-	// SourcePath contains the path where this job is defined
-	SourcePath string `json:"-"`
+	// SourcePath contains the path where the PipelineRun YAML is defined
+	SourcePath string `json:"sourcePath,omitempty"`
 	// Spec is the Kubernetes pod spec used if Agent is kubernetes.
 	Spec *v1.PodSpec `json:"spec,omitempty"`
 	// PipelineRunSpec is the Tekton PipelineRun spec used if agent is tekton-pipeline
 	PipelineRunSpec *tektonv1beta1.PipelineRunSpec `json:"pipelineRunSpec,omitempty"`
 	// PipelineRunParams are the params used by the pipeline run
-	PipelineRunParams []job.PipelineRunParam `json:"pipeline_run_params,omitempty"`
+	PipelineRunParams []PipelineRunParam `json:"pipelineRunParams,omitempty"`
 
 	UtilityConfig
 }
 
+// PipelineRunParam represents a param used by the pipeline run
+type PipelineRunParam struct {
```

Comment:
Why do we need to duplicate the struct here ?

Reply:
to use tekton style YAML notation rather than prow style YAML (so the trigger file + tekton pipelines look and feel like they are designed to work together & part of the k8s ecosystem)

Reply:
yeah but why not reuse the existing structs ? is this necessary to duplicate ?

Reply:
I was trying to avoid the `always_run` style of YAML marshalling that the shared prow config uses to use k8s camelCase naming conventions like k8s / tekton / istio etc use such as `alwaysRun`. It adds complexity and confusion though; so I've just removed them all - we're now just using `job.Presubmit` / `job.Postsubmit`directly

Turns out presubmit / postsubmits are fairly simple and there's not many multiple-word fields so its probably not that big of a UX fail really. Longer term we could try move to shorter, simpler property names in Presubmit/Postsubmits maybe anyway ;) (while having a parser to handle the older names). e.g. `always_run` -> `always` and `rerun_command` -> `rerun` maybe?